### ISSUE TITLE
Built-in fonts

### ISSUE THREAD

**Issue by nagyrobi** [clydebarrow/esphome#48]
```
Compiling .pioenvs/esp32-test-lvgl-host/src/main.o
<unicode string>: In function ‘void setup()’:
<unicode string>:2204:62: error: ‘montserrat_8’ was not declared in this scope
*** [.pioenvs/esp32-test-lvgl-host/src/main.o] Error 1
```
and 
```
Compiling .pioenvs/esp32-test-lvgl-host/src/main.o
<unicode string>: In function ‘void setup()’:
<unicode string>:2204:62: error: ‘montserrat_10’ was not declared in this scope
*** [.pioenvs/esp32-test-lvgl-host/src/main.o] Error 1
```
and
```
Compiling .pioenvs/esp32-test-lvgl-host/src/main.o
<unicode string>: In function ‘void setup()’:
<unicode string>:2204:62: error: ‘simsun_16_cjk’ was not declared in this scope
*** [.pioenvs/esp32-test-lvgl-host/src/main.o] Error 1
```

although they are compiled:
```
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_dejavu_16_persian_hebrew.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_fmt_txt.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_loader.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_10.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_12.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_12_subpx.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_14.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_16.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_18.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_20.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_22.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_24.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_26.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_28.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_28_compressed.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_30.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_32.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_34.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_36.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_38.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_40.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_42.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_44.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_46.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_48.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_montserrat_8.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_simsun_16_cjk.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_unscii_16.o
Compiling .pioenvs/esp32-test-lvgl-host/lib09a/lvgl/font/lv_font_unscii_8.o
```
Not sure what `fmt_txt` and `loader` fonts are, but they are compiled, would probably worth to try and see.

---
with `text_font: montserrat_28_compressed` labels are empty, and the LVGL logs this:

```
[I][lvgl:000]: [Warn]   (397413.594, +0)         lv_font_get_bitmap_fmt_txt: Compressed fonts is used but LV_USE_FONT_COMPRESSED is not enabled in lv_conf.h  (in lv_font_fmt_txt.c line #131)
[I][lvgl:000]: [Warn]   (397413.594, +0)         lv_draw_sw_letter: lv_draw_letter: character's bitmap not found        (in lv_draw_sw_letter.c line #145)
```

with `text_font: montserrat_12_subpx` labels are empty, and the LVGL logs this:
```
[I][lvgl:000]: [Warn]   (397578.880, +0)         lv_draw_sw_letter: Can't draw sub-pixel rendered letter because LV_USE_FONT_SUBPX == 0 in lv_conf.h    (in lv_draw_sw_letter.c line #153)
```

**Comment by nagyrobi** [comment]
OK, after reading a bit, `compressed` and `subpx` can be ditched.

TLDR;
https://docs.lvgl.io/8/overview/font.html#subpixel-rendering this looks as something pretty hardware-specific:

> Subpixel rendering works only if the color channels of the pixels have a horizontal layout. That is the R, G, B channels are next each other and not above each other. 

You can't rotate the screen because of this. I guess this is targeted in LVGL to speciality hardware which is developed with a certain screen in mind.

We could ditch `montserrat_12_subpx`, also because this subpixel font is only available in12px size. It also requires special parameters at compile time. Imho if anyone wants to play with subpixel rendering in ESPHome, should try on a dedicated branch with custom fonts rendered.

https://docs.lvgl.io/8/overview/font.html#compress-fonts compressed fonts have a compromise:

> The compression is more effective with larger fonts and higher bpp. However, it's about 30% slower to render the compressed fonts. Therefore it's recommended to compress only the largest fonts of user interface, because
> 
> - they need the most memory
> - they can be compressed better
> - and probably they are used less frequently then the medium sized fonts, so the performance cost is smaller.

Don't think this is appropriate for ESP's, or the platforms ESPHome targets. I thinbk `montserrat_28_compressed` is the only one included (just like `montserrat_12_subpx`) just to demonstrate that these featrues are doable in LVGL, but imho ESPHome doesn't need these.

**Comment by clydebarrow** [comment]
Done
